playbooks/azure/openshift-cluster/build_image.yml

```diff
@@ -0,0 +1,126 @@
+---
+# Note: this required ansible-galaxy install Azure.azure_modules
+
+- import_playbook: prerequisites.yml
+
+# create all the items necessary to make the VM
+- name: "Create resource group"
+  hosts: localhost
+  gather_facts: yes
+  tasks:
+  - name: provision public ip address
+    import_role:
+      name: openshift_azure
+      tasks_from: provision_instance.yml
+    vars:
+      openshift_azure_vm_base_image:
+        name: '{{ openshift_azure_base_image }}'
+        resource_group: '{{ openshift_azure_base_image_ns }}'
+  - name: get the openshift registry token
+    import_role:
+      name: openshift_registry_token
+    when:
+    - oreg_auth_password is not defined
+    - openshift_registry_token_api_auth_password is defined
+
+- name: prepare node image
+  hosts: nodes
+  gather_facts: yes
+  remote_user: "{{ openshift_azure_vm_admin_username }}"
+  become: True
+  tasks:
+  - name: read the openshift registry token from localhost's vars
+    set_fact:
+      oreg_url: '{{ openshift_registry_token_url }}'
+      oreg_auth_user: '{{ openshift_registry_token_auth_user }}'
+      oreg_auth_password: '{{ hostvars["localhost"]["openshift_registry_token_auth_password"] }}'
+    when:
+    - oreg_auth_password is not defined
+    - openshift_registry_token_api_auth_password is defined
+  - name: set the user to perform installation
+    set_fact:
+      ansible_ssh_user: "{{ openshift_azure_vm_admin_username | default(ansible_ssh_user) }}"
+      openshift_node_bootstrap: True
+      openshift_node_image_prep_packages:
+      - cloud-utils-growpart
+      ansible_become: true
+
+  - name: lay down yum certs
+    copy:
+      content: "{{ item.content }}"
+      dest: "/var/lib/yum/{{ item.name }}"
+    with_items:
+    - name: client-cert.pem
+      content: "{{ yum_client_cert_contents }}"
+    - name: client-key.pem
+      content: "{{ yum_client_key_contents }}"
```

Comment:
setting mode 0600 on this might be a useful additional protection

Reply:
done